apps/ledger-live-mobile/src/newArch/features/WalletSync/screens/Activation/ActivationDrawer.tsx

```diff
@@ -1,37 +1,57 @@
-import React from "react";
+import React, { useState } from "react";
 import QueuedDrawer from "LLM/components/QueuedDrawer";
-import Activation from "../../components/Activation";
 import { TrackScreen } from "~/analytics";
-import ChooseSyncMethod from "../Synchronize/ChooseMethod";
+import { useWindowDimensions } from "react-native";
+import { Flex } from "@ledgerhq/native-ui";
+import ActivationFlow from "../../components/Activation/ActivationFlow";
+import { Steps } from "../../types/Activation";
+import DrawerHeader from "../../components/Synchronize/DrawerHeader";
 
 type Props = {
   isOpen: boolean;
-  reopenDrawer: () => void;
+  startingStep: Steps;
   handleClose: () => void;
 };
 
-const ActivationDrawer = ({ isOpen, handleClose, reopenDrawer }: Props) => {
-  const [isSyncMethodDrawerOpen, setIsSyncMethodDrawerOpen] = React.useState(false);
+const ActivationDrawer = ({ isOpen, startingStep, handleClose }: Props) => {
+  const [currentStep, setCurrentStep] = useState<Steps>(startingStep);
+  const { height } = useWindowDimensions();
+  const maxDrawerHeight = height - 180;
 
-  const onPressCloseDrawer = () => {
-    setIsSyncMethodDrawerOpen(false);
-    reopenDrawer();
-  };
+  const CustomDrawerHeader = () => <DrawerHeader onClose={handleClose} />;
+
+  const handleStepChange = (step: Steps) => setCurrentStep(step);
+
+  let goBackCallback: () => void;
+
+  const hasCustomHeader = currentStep === Steps.QrCodeMethod && startingStep === Steps.Activation;
 
-  const openSyncMethodDrawer = () => {
-    setIsSyncMethodDrawerOpen(true);
+  const canGoBack = currentStep === Steps.ChooseSyncMethod && startingStep === Steps.Activation;
+
+  const resetStep = () => setCurrentStep(startingStep);
+
+  const onClose = () => {
+    resetStep();
     handleClose();
   };
 
   return (
     <>
       <TrackScreen />
-      <QueuedDrawer isRequestingToBeOpened={isOpen} onClose={handleClose}>
-        <Activation isInsideDrawer openSyncMethodDrawer={openSyncMethodDrawer} />
-      </QueuedDrawer>
-
-      <QueuedDrawer isRequestingToBeOpened={isSyncMethodDrawerOpen} onClose={onPressCloseDrawer}>
-        <ChooseSyncMethod />
+      <QueuedDrawer
+        isRequestingToBeOpened={isOpen}
+        onClose={onClose}
+        CustomHeader={hasCustomHeader ? CustomDrawerHeader : undefined}
+        hasBackButton={canGoBack}
+        onBack={() => goBackCallback()}
+      >
+        <Flex maxHeight={maxDrawerHeight}>
+          <ActivationFlow
+            startingStep={currentStep}
+            onStepChange={handleStepChange}
+            onGoBack={callback => (goBackCallback = callback)}
```

Comment:
you should try using maybe refs

Reply:
I think ref would make this more complex to read

Reply:
I think that we have this issue because we manage the different drawers in the Activation Flow and should be at the top level. Is it possible to do it that way or do we have any limitation?

Reply:
It's possible I did it that way so I could call the flow without being in a drawer, but that's probably never going to happen

Reply:
✅